plugin-core/src/main/java/appland/settings/AppMapApplicationSettings.java

```diff
@@ -31,6 +33,13 @@ public class AppMapApplicationSettings {
      */
     private volatile boolean showFirstAppMapNotification = false;
 
+    /**
+     * Map of environment variables to be set when starting AppMap services
+     * The key is the name of the environment variable and the value is the value of the environment variable.
+     */
+    private volatile Map<String, String> cliEnvironment = new HashMap<>();
```

Comment:
I'd prefer if you override the setter of this map to avoid re-assigning the field.
Currently, `apply` of the settings is re-assigning the map to an unknown map implementation, which may cause problems with serialization.
With our own setter we could always use map implementation it's considering safe.

If the setter is overridden, Kotlin will automatically use it.

```
public void setCliEnvironment(Map<String, String> cliEnvironment) {
        this.cliEnvironment = Map.copyOf(cliEnvironment); // unmodifiable map
    }
```

Reply:
Yup, good idea, thanks!